packages/coinstac-simulator/src/index.js

```diff
@@ -32,6 +34,92 @@ const processes = {
 
 const exportList = {
   /**
+   * Get a declaration object.
+   * @private
+   *
+   * @param {string} declarationPath
+   * @returns {Promise} Resolves to a declaration object
+   */
+  getDeclaration(declarationPath) {
+    if (typeof declarationPath !== 'string') {
+      return Promise.reject(new Error('Requires a declaration path string'));
+    }
+
+    return bluebird.promisify(fs.stat)(declarationPath)
+      .then(stats => {
+        if (!stats.isFile() && !stats.isDirectory()) {
+          throw new Error(`Couldn't find declaration ${declarationPath}`);
+        }
+
+        /* eslint-disable global-require */
+        const declaration = require(declarationPath);
+        /* eslint-ensable global-require */
+
+        // Validate declaration's shape
```

Comment:
I chose native JavaScript checks so we can throw concise errors. Joi errors are guaranteed to confuse our mostly JavaScript newbie audience.